ADAL/src/telemetry/ADTelemetry.m

```diff
@@ -177,14 +177,21 @@ - (void)dispatchEventNow:(NSString*)requestId
     {
         for (ADDefaultDispatcher *dispatcher in _dispatchers)
         {
-            for (NSString *propertyName in [event.propertyMap allKeys]) {
+            id<ADTelemetryEventInterface>eventCopy = [event copyWithZone:nil];
+            
+            for (NSString *propertyName in [eventCopy.propertyMap allKeys]) {
                 BOOL isPii = [ADTelemetryPiiRules isPii:propertyName];
                 if (isPii && !self.piiEnabled) {
-                    [event deleteProperty:propertyName];
+                    [eventCopy deleteProperty:propertyName];
+                } else if (isPii && self.piiEnabled) {
+                    NSString *value = eventCopy.propertyMap[propertyName];
+                    if (![NSString adIsStringNilOrBlank:value]) {
+                        [eventCopy setProperty:propertyName value:[value adComputeSHA256]];
```

Comment:
Don't add hashing here, add it in the setter. We already have multiple fields getting hashed in the setter, let's keep it consistent.

Reply:
Agreed it should be done in the setter.